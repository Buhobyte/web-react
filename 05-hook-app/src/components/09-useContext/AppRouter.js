import React from 'react'
// https://v5.reactrouter.com/web/guides/quick-start
import {
    BrowserRouter as Router,
    Routes,
    Route,
    Navigate
} from "react-router-dom";


import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className='container'>
                    <Routes>
                        <Route path="/" element={<HomeScreen />}></Route>
                        <Route path="/about" element={<AboutScreen />}></Route>
                        <Route path="/login" element={<LoginScreen />}></Route>
                        {/* <Route component={HomeScreen} /> */}
                        <Route path="*" element={<Navigate to="/" />} />
                    </Routes>
                </div>
            </div>
        </Router>
    )
}
