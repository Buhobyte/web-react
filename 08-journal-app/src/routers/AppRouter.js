import React, { useEffect, useState } from 'react'
import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom'
import { AuthRouter } from './AuthRouter';
import { JournalScreen } from './../components/journal/JournalScreen';
import { onAuthStateChanged, getAuth } from 'firebase/auth';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';
import { PrivateRoute } from '../reducers/PrivateRoute';
import { PublicRoute } from './../reducers/PublicRoute';
import { startLoadingNotes } from './../actions/notes';

export const AppRouter = () => {

    const dispatch = useDispatch();

    const [checking, setCheking] = useState(true);
    const [isLogginIn, setIsLogginIn] = useState(false);

    useEffect(() => {
        onAuthStateChanged(getAuth(), async (user) => {

            if (user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsLogginIn(true);

                dispatch(startLoadingNotes(user.uid));

            } else {
                setIsLogginIn(false);
            }

            setCheking(false);
        })
    }, [dispatch, setCheking, isLogginIn])


    if (checking) {
        return <div>Checking...</div>
    }


    return (
        <BrowserRouter>
            <Routes>
                <Route path="/" element={

                    <PrivateRoute isAuthenticated={isLogginIn}>
                        <JournalScreen />
                    </PrivateRoute>

                } />
                <Route path="/auth/*" element={
                    <PublicRoute isAuthenticated={isLogginIn}>
                        <AuthRouter />
                    </PublicRoute>
                } />
                <Route path="*" element={<Navigate to="/auth/login" replace />} />
            </Routes>
        </BrowserRouter>
    )
}
