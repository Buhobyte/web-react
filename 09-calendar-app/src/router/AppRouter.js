
import React from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { CalendarScreen } from './../components/calendar/CalendarScreen';
import { LoginScreen } from './../components/auth/LoginScreen';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path="/" element={<CalendarScreen />} />
                <Route path="login" element={<LoginScreen />} />
            </Routes>
        </BrowserRouter>
    )
}
