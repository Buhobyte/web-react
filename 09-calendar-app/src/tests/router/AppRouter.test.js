import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom'
import { AppRouter } from './../../router/AppRouter';

const mockStore = configureStore([thunk]);

const initialState = {
    auth: {
        checking: false,
    }
}
const store = mockStore(initialState);
// store.dispatch = jest.fn();

describe('Pruebas en <AppRouter />', () => {


    test('debe de motrar el espere...', () => {

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>)

        expect(wrapper).toMatchSnapshot();

    });
});