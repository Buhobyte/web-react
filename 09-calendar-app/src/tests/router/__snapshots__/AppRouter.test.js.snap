// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de motrar el espere... 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3>
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control"
            name="lEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="text"
            value=""
          />
        </div>
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control"
            name="lPassword"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="input-group mb-3"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
    <div
      className="col-md-6 login-form-2"
    >
      <h3>
        Registro
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control"
            name="rName"
            onChange={[Function]}
            placeholder="Nombre"
            type="text"
            value=""
          />
        </div>
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control"
            name="rEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="email"
            value=""
          />
        </div>
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control"
            name="rPassword"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control"
            name="rConfirmPassword"
            onChange={[Function]}
            placeholder="Repita la contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="input-group mb-3"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Crear cuenta"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
